crates/orchestrator/src/config.rs

```diff
@@ -208,27 +208,75 @@ impl<K: SignatureKey, E: ElectionConfig> NetworkConfig<K, E> {
                     // fallback to orchestrator
                     error!("{e}, falling back to orchestrator");
 
-                    let config = client.get_config(client.identity.clone()).await;
+                    let config = client
+                        .get_config_without_peer(client.identity.clone())
+                        .await;
 
                     // save to file if we fell back
                     if let Err(e) = config.to_file(file) {
                         error!("{e}");
                     };
 
-                    (config, NetworkConfigSource::Orchestrator)
+                    (config, NetworkConfigSource::File)
                 }
             }
         } else {
             error!("Retrieving config from the orchestrator");
 
             // otherwise just get from orchestrator
             (
-                client.get_config(client.identity.clone()).await,
+                client
+                    .get_config_without_peer(client.identity.clone())
+                    .await,
                 NetworkConfigSource::Orchestrator,
             )
         }
     }
 
+    /// Get a temporary node index for generating a validator config
+    pub async fn generate_init_validator_config(client: &OrchestratorClient) -> ValidatorConfig<K> {
+        // This cur_node_index is only used for key pair generation, it's not bound with the node,
+        // lather the node with the generated key pair will get a new node_index from orchestrator.
+        let cur_node_index = client.get_node_index_for_init_validator_config().await;
+        ValidatorConfig::generated_from_seed_indexed([0u8; 32], cur_node_index.into(), 1)
+    }
+
+    /// Asynchronously retrieves a `NetworkConfig` from an orchestrator.
+    /// The retrieved one includes correct `node_index` and peer's public config.
+    pub async fn get_complete_config(
+        client: &OrchestratorClient,
+        my_own_validator_config: ValidatorConfig<K>,
+        file: Option<String>,
```

Comment:
Just FYI, I think the sequencer is _always_ going to use this with `file: None`, loading the config from the orchestrator. We manually check if the config already exists before calling this function, because depending on the sequencer config, we may check either a file or Postgres, which Hotshot doesn't support.

I think this is fine though, looking at the logic of this function. Setting `file` to `None` basically just means "get this config from the orchestrator" which is exactly what we want